src/properties.ts

```diff
@@ -0,0 +1,277 @@
+/**
+ * Property field types
+ * examples are propvided for the different types:
```

Comment:
Typo: `provided`

Reply:
> I mainly added comments about typos ;). As a TS point of view, it LGTM and for the naming question I didn't use `knob` enough to have a strong opinion on it.

Thanks a lot, fixing them one by one and will mark as resolved